crates/l2/operator/l1_tx_sender.rs

```diff
@@ -1,23 +1,35 @@
-use crate::utils::eth_client::EthClient;
+use crate::utils::{
+    config::{eth::EthConfig, l1_tx_sender::L1TxSenderConfig},
+    eth_client::EthClient,
+};
 use ethereum_rust_blockchain::constants::TX_GAS_COST;
 use ethereum_rust_core::types::{EIP1559Transaction, TxKind};
 use ethereum_types::{Address, H256};
 use libsecp256k1::SecretKey;
-use std::str::FromStr;
 use tracing::{debug, warn};
 
-const RICH_WALLET_PK: &str = "0x385c546456b6a603a1cfcaa9ec9494ba4832da08dd6bcf4de9a71e4a01b74924";
-const RICH_WALLET_ADDR: &str = "0x3D1e15a1a55578f7c920884a9943b3B35D0D885b";
-const BLOCK_EXECUTOR_ADDR: &str = "0x31e68fE377E509c8324b6206ADC7f11003Bd9130";
 const COMMIT_FUNCTION_SELECTOR: [u8; 4] = [227, 206, 9, 77];
 const VERIFY_FUNCTION_SELECTOR: [u8; 4] = [142, 118, 10, 254];
 
-pub struct L1TxSender;
+pub struct L1TxSender {
+    rpc_url: String,
+    contract_address: Address,
+    operator_address: Address,
+    operator_private_key: SecretKey,
+}
 
 impl L1TxSender {
-    async fn send_transaction(mut tx: EIP1559Transaction) -> Result<H256, String> {
-        let client = EthClient::new("http://localhost:8545");
-        let private_key = SecretKey::parse(&H256::from_str(RICH_WALLET_PK).unwrap().0).unwrap();
+    pub fn new_from_config(sender_config: L1TxSenderConfig, eth_config: EthConfig) -> Self {
```

Comment:
An alternative to this is to implement `From<L1TxSenderConfig> for L1TxSender`, and for all the components with the same behavior. It feels like the same for me, what do you prefer?

Reply:
Oh didn't notice this. Let's try to do a 1 to 1 mapping. In the case of the repeated config we can do something I'll suggest below.

Reply:
Nvm, we'll leave it for another instance